### ISSUE TITLE
Resilience patterns APIs are non consistent

### ISSUE THREAD

**Issue by FSeidinger** [resilience4j/resilience4j#1949]
Resilience4j version: 2.02

Java version: Eclipse Adoptium 17.0.2.8

The APIs for the resilience patterns, given by static methods on the interfaces like `CircuitBreaker ` or `Retry` , is not consistent over the different APIs.

For example the `CircuitBreaker` let's decorate a `Consumer` with `decorateConsumer(CircuitBreaker circuitBreaker, Consumer<T> consumer)` whereas in the `Retry` this method is missing and you cannot decorate a Consumer.

Also there is no easy way to decorate a service call having no return value (void) and multiple arguments. consider the following example:

```Java
public interface Storage {
    void save(final String key, final String value) throws StorageException;
}
```

**Comment by RobWin** [comment]
Hi,

yes, the Consumer interface is missing, because it does not really provide any advantage.


simplest solution might be to use `resilience4j-all`.

```` 
Decorators.ofCheckedRunnable(() -> storave.save(key, value))
      .withCircuitBreaker(circuitBreaker)
      .withRetry(retry)
      .run();
```

**Comment by RobWin** [comment]
Yes, that's correct.
We didn't want to provide a decorator for all functions, like CheckedFunction1, CheckedFunction2, and so on.
The solution just to provide a decorator for Runnable or Callable seemed to be fine.

**Comment by FSeidinger** [comment]
Thanks for the clarification and having a strong opinion on that. As an architect I like it, although it might be slightly disappointing for the users :smiley:

**Comment by nikolayandr** [comment]
> https://github.com/resilience4j/resilience4j/issues/1949

@RobWin totally disaggree, Consumer is useful when it is used out of scope where it was defined, so it is not possible to define retry-able consumer only once and reuse it by multiply different processes,  and the second point: 
https://resilience4j.readme.io/docs/retry claims Consumer and CheckedConsumer is supported by Retry:

```Decorate and execute a functional interface

As you can guess Retry has all sort of higher order decorator functions just like CircuitBreaker. You can decorate any Callable, Supplier, Runnable, Consumer, CheckedRunnable, CheckedSupplier, CheckedConsumer or CompletionStage with a Retry.

**Comment by RobWin** [comment]
Hallo, 
please feel free to create a PR. We really welcome contributions.

**Comment by nikolayandr** [comment]
> Hallo, please feel free to create a PR. We really welcome contributions. I never declined to add it. I just didnt have time to do it.

Yes sir :) 

@FSeidinger https://github.com/resilience4j/resilience4j/pull/2249 should solve this issue

**Comment by RobWin** [comment]
Thank you
